### PR TITLE
add TF32 sample code

### PR DESCRIPTION
**Pull request by tsenwang**

### CODE REVIEW

**Comment by tsenwang** [comment]
root@banff-cyxtera-s73-1:/Workspace/hipBLASLt/build/release# make test
Running tests...
Test project /Workspace/hipBLASLt/build/release
      Start  1: sample_hipblaslt_gemm
 1/39 Test  #1: sample_hipblaslt_gemm ...............................   Passed    1.47 sec
      Start  2: sample_hipblaslt_gemm_ext
 2/39 Test  #2: sample_hipblaslt_gemm_ext ...........................   Passed    1.48 sec
      Start  3: sample_hipblaslt_gemm_batched
 3/39 Test  #3: sample_hipblaslt_gemm_batched .......................   Passed    1.27 sec
      Start  4: sample_hipblaslt_gemm_batched_ext
 4/39 Test  #4: sample_hipblaslt_gemm_batched_ext ...................   Passed    2.23 sec
      Start  5: sample_hipblaslt_gemm_tuning_splitk_ext
 5/39 Test  #5: sample_hipblaslt_gemm_tuning_splitk_ext .............   Passed   10.36 sec
      Start  6: sample_hipblaslt_gemm_bias
 6/39 Test  #6: sample_hipblaslt_gemm_bias ..........................   Passed    1.01 sec
      Start  7: sample_hipblaslt_gemm_bias_ext
 7/39 Test  #7: sample_hipblaslt_gemm_bias_ext ......................   Passed    1.00 sec
      Start  8: sample_hipblaslt_gemm_get_all_algos
 8/39 Test  #8: sample_hipblaslt_gemm_get_all_algos .................   Passed   11.21 sec
      Start  9: sample_hipblaslt_gemm_get_all_algos_ext
 9/39 Test  #9: sample_hipblaslt_gemm_get_all_algos_ext .............   Passed   10.57 sec
      Start 10: sample_hipblaslt_gemm_get_algo_by_index_ext
10/39 Test #10: sample_hipblaslt_gemm_get_algo_by_index_ext .........   Passed    9.27 sec
      Start 11: sample_hipblaslt_gemm_alphavec_ext
11/39 Test #11: sample_hipblaslt_gemm_alphavec_ext ..................   Passed    1.32 sec
      Start 12: sample_hipblaslt_gemm_gelu_aux_bias
12/39 Test #12: sample_hipblaslt_gemm_gelu_aux_bias .................   Passed    1.01 sec
      Start 13: sample_hipblaslt_gemm_gelu_aux_bias_ext
13/39 Test #13: sample_hipblaslt_gemm_gelu_aux_bias_ext .............   Passed    0.67 sec
      Start 14: sample_hipblaslt_gemm_amax
14/39 Test #14: sample_hipblaslt_gemm_amax ..........................   Passed    1.05 sec
      Start 15: sample_hipblaslt_gemm_amax_ext
15/39 Test #15: sample_hipblaslt_gemm_amax_ext ......................   Passed    0.68 sec
      Start 16: sample_hipblaslt_gemm_amax_with_scale
16/39 Test #16: sample_hipblaslt_gemm_amax_with_scale ...............   Passed    1.07 sec
      Start 17: sample_hipblaslt_gemm_amax_with_scale_ext
17/39 Test #17: sample_hipblaslt_gemm_amax_with_scale_ext ...........   Passed    1.02 sec
      Start 18: sample_hipblaslt_gemm_bgradb
18/39 Test #18: sample_hipblaslt_gemm_bgradb ........................   Passed    0.85 sec
      Start 19: sample_hipblaslt_gemm_ext_bgradb
19/39 Test #19: sample_hipblaslt_gemm_ext_bgradb ....................   Passed    0.86 sec
      Start 20: sample_hipblaslt_gemm_dgelu_bgrad
20/39 Test #20: sample_hipblaslt_gemm_dgelu_bgrad ...................   Passed    0.88 sec
      Start 21: sample_hipblaslt_gemm_dgelu_bgrad_ext
21/39 Test #21: sample_hipblaslt_gemm_dgelu_bgrad_ext ...............   Passed    0.88 sec
      Start 22: sample_hipblaslt_gemm_is_tuned_ext
22/39 Test #22: sample_hipblaslt_gemm_is_tuned_ext ..................   Passed    1.03 sec
      Start 23: sample_hipblaslt_gemm_tuning_wgm_ext
23/39 Test #23: sample_hipblaslt_gemm_tuning_wgm_ext ................   Passed   15.23 sec
      Start 24: sample_hipblaslt_gemm_with_scale_a_b
24/39 Test #24: sample_hipblaslt_gemm_with_scale_a_b ................   Passed   15.01 sec
      Start 25: sample_hipblaslt_gemm_with_scale_a_b_ext
25/39 Test #25: sample_hipblaslt_gemm_with_scale_a_b_ext ............   Passed    0.85 sec
      Start 26: sample_hipblaslt_groupedgemm_ext
26/39 Test #26: sample_hipblaslt_groupedgemm_ext ....................   Passed    2.45 sec
      Start 27: sample_hipblaslt_groupedgemm_fixed_mk_ext
27/39 Test #27: sample_hipblaslt_groupedgemm_fixed_mk_ext ...........   Passed   16.09 sec
      Start 28: sample_hipblaslt_groupedgemm_get_all_algos_ext
28/39 Test #28: sample_hipblaslt_groupedgemm_get_all_algos_ext ......   Passed   13.60 sec
      Start 29: sample_hipblaslt_gemm_mix_precision
29/39 Test #29: sample_hipblaslt_gemm_mix_precision .................   Passed    0.77 sec
      Start 30: sample_hipblaslt_gemm_mix_precision_ext
30/39 Test #30: sample_hipblaslt_gemm_mix_precision_ext .............   Passed    1.73 sec
      Start 31: sample_hipblaslt_gemm_mix_precision_with_amax_ext
31/39 Test #31: sample_hipblaslt_gemm_mix_precision_with_amax_ext ...   Passed    1.16 sec
      Start 32: sample_hipblaslt_gemm_attr_tciA_tciB
32/39 Test #32: sample_hipblaslt_gemm_attr_tciA_tciB ................   Passed    0.97 sec
      Start 33: sample_hipblaslt_ext_op_layernorm
33/39 Test #33: sample_hipblaslt_ext_op_layernorm ...................   Passed    0.56 sec
      Start 34: sample_hipblaslt_gemm_ext_deprecated
34/39 Test #34: sample_hipblaslt_gemm_ext_deprecated ................   Passed    1.47 sec
      Start 35: sample_hipblaslt_ext_op_amax
35/39 Test #35: sample_hipblaslt_ext_op_amax ........................   Passed    0.78 sec
      Start 36: sample_hipblaslt_ext_op_amax_with_scale
36/39 Test #36: sample_hipblaslt_ext_op_amax_with_scale .............   Passed    0.57 sec
      Start 37: sample_hipblaslt_gemm_with_TF32
37/39 Test #37: sample_hipblaslt_gemm_with_TF32 .....................   Passed    0.72 sec
      Start 38: hipblaslt-bench-groupedgemm-fixed-mk
38/39 Test #38: hipblaslt-bench-groupedgemm-fixed-mk ................   Passed  240.64 sec
      Start 39: hipblaslt-test
39/39 Test #39: hipblaslt-test ......................................   Passed  1009.49 sec

100% tests passed, 0 tests failed out of 39

Total Test time (real) = 1383.30 sec

**Review by hcman2 — DISMISSED** [review]

File: clients/samples/25_gemm_with_TF32/sample_hipblaslt_gemm_with_TF32.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                hipStream_t        stream);
+
+int main()
+{
+    /** This is a NN example with
+     *  a = (m, k). lda = m
+     *  b = (k, n). ldb = k
+     *  c = d = (m, n). ldc = ldd = m
+     */
+    Runner<hipblasLtHalf, hipblasLtHalf, hipblasLtHalf, float, float> runner(
```
Comment (jichangjichang) [inline clients/samples/25_gemm_with_TF32/sample_hipblaslt_gemm_with_TF32.cpp]: fix the data type

File: clients/samples/25_gemm_with_TF32/sample_hipblaslt_gemm_with_TF32.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                                                          matC,
+                                                          matD,
+                                                          pref,
+                                                          request_solutions,
+                                                          heuristicResult,
+                                                          &returnedAlgoCount));
+
+    if(returnedAlgoCount == 0)
+    {
+        std::cerr << "No valid solution found!" << std::endl;
```
Comment (jichangjichang) [inline clients/samples/25_gemm_with_TF32/sample_hipblaslt_gemm_with_TF32.cpp:165]: Destroy descriptors before return

**Review by jichangjichang — APPROVED** [review]

**Review by vin-huang — APPROVED** [review]
LGTM

**Pull request merged**